src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -153,6 +153,10 @@ public function decode($data, $format, array $context = array())
         }
         fclose($handle);
 
+        if (isset($context['alwaysAsCollection']) && true === $context['alwaysAsCollection']) {
```

Comment:
The key should be in snake case, as for all other context’s keys.
By the way, can't it just be `collection`?